### PR TITLE
170947380 - Users Should Receive  Reset Password Link

### PR DESCRIPTION
**Pull request by Moise1**
#### What does this PR do?
It allows a user to receive a link in their email address to reset their password.

#### Description of Task to be completed?

- send a password reset  link to the user's email address

#### How should this be manually tested?
-  Run this `git clone https://github.com/Stackup-Rwanda/octopus-bn-frontend.git` in your terminal if you don't have the project on your computer
- If you have it, run `git fetch && git checkout ft-reset-password-170947380` instead
- Run `npm run dev` command 
- In your browser go to `localhost:your_port/forgot-password`
- Enter an existing and accessible email in the field
- Go that same email address and you'll see the rese password email.

##### Reset Password  Request Page

Screen Shot 2020-04-27 at 12 18 14

#### What are the relevant pivotal tracker stories?
[#170947380](https://www.pivotaltracker.com/story/show/170947380)

### CODE REVIEW

File: src/__tests__/login.spec.js [hunk 1]
```diff
@@ -1,113 +1,144 @@
-import React from 'react';
```
Comment (blaise82) [inline src/__tests__/login.spec.js]: i think something isn't right with your prettier

File: src/components/SendResetEmail.js [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+			[e.target.name]: e.target.value
+		});
+	};
+	render() {
+		return (
+			<div>
+				<div className={styles.forms}>
+					<div style={{ display: this.props.sendEMailState.loading }}>
+						<LinearProgress />
+					</div>
```
Comment (blaise82) [inline src/components/SendResetEmail.js:77]: please add the change language buttons

File: src/views/PasswordResetProcess.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+          <p className={styles.fs_confirm} style={{ marginTop: "360px" }}>
+            Please go to your email address to complete the process.
+          </p>
+        </div>
+      </div>
+    </div>
+  );
+};
+
+export default PasswordResetProcess;
```
Comment (blaise82) [inline src/views/PasswordResetProcess.js]: you didn't have to create a new view just to display a message

File: webpack.config.js [hunk 4]
```diff
@@ -29,7 +29,7 @@ module.exports = {
         use: ["style-loader", "css-loader", "sass-loader"],
       },
       {
-        test: /\.css$/,
+        test: /\.(|scss|sass|css)$/,
```
Comment (blaise82) [inline webpack.config.js]: we had this to handle other files
```

      {
        test: /\.s[ac]ss$/i,
        use: ["style-loader", "css-loader", "sass-loader"],
      },
```

are you have trouble loading .css files or sass?

File: src/components/SendResetEmail.js (hunk 2 above)
Comment (blaise82) [reply src/components/SendResetEmail.js:77]: just noticed that you do have them.
please update your screenshot

File: webpack.config.js (hunk 4 above)
Comment (Moise1) [reply webpack.config.js]: Yes! For some reasons, I wasn't able to run build script due to the .scss issues. So, I added that.

**Comment by AbdoulNuru** [comment]
@Moise1 after a user receive an email, I realized they really can't change their passwords

File: src/__tests__/login.spec.js (hunk 1 above)
Comment (Moise1) [reply src/__tests__/login.spec.js]: What do you mean?

File: src/__tests__/ActionsTest.spec.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        { type: types.SEND_EMAIL_SUCCESS, payload: {email: 'moiseri01@gmail.com' } }
+      ]
+      const store = mockStore({ email: 'moiseri01@gmail.com' })
+  
+      return store.dispatch(actions.sendMailAction()).then(() => {
+        // return of async actions
+        expect(store.getState()).toEqual(expectedActions[1].payload)
+      })
+    })
+})
```
Comment (hakoemmy) [inline src/__tests__/ActionsTest.spec.js]:
```suggestion
})

```
Add a new line

File: src/routes/index.js [hunk 6]
```diff
@@ -13,6 +17,9 @@ export default class index extends Component {
         <Route path="/dashboard" exact component={Dashboard} />
         <Route path="/signup" exact component={Signup} />
         <Route path="/verify-email" exact component={Confirm} />
+        <Route path="/forgot-password" exact component={SendResetEmailView}/>
+        <Route path="/password-reset-process" exact component={PasswordResetProcess}/>
```
Comment (hakoemmy) [inline src/routes/index.js]: @Moise1  Just a suggestion, can't you call ```/password-reset-process``` something short and clear? Take a look at my suggestion

```suggestion
        <Route path="/password-reset" exact component={PasswordResetProcess}/>
```

Comment (Moise1) [reply src/routes/index.js:21]: @AbdoulNuru , yeah! A user cannot change the password just yet. The story that I'm working on has a main action of sending the link to the user.

File: src/views/PasswordResetProcess.js (hunk 3 above)
Comment (Moise1) [reply src/views/PasswordResetProcess.js]: @blaise82, I'm afraid that the styling won't be the same.

**Comment by Moise1** [comment]
> > @Moise1 after a user receive an email, I realized they really can't change their passwords
> 
> @AbdoulNuru , yeah! A user cannot change the password just yet. The story that I'm working on has a main action of sending the link to the user.
> 

I user should be able to change their password.

File: src/routes/index.js (hunk 6 above)
Comment (Moise1) [reply src/routes/index.js]: @hakoemmy , I'd like to keep it that way coz the process isn't complete until a user provides a new password.

**Pull request closed without merging**